### PR TITLE
Bug 1390870 - Implement API endpoint for a worker

### PR DESCRIPTION
**Pull request by helfi92**
API endpoint to `GET` and `PUT` a worker

### CODE REVIEW

File: src/api.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ let Promise     = require('promise');
 
 // Maximum number runs allowed
 const MAX_RUNS_ALLOWED = 50;
-
+const SLUGID_SIZE = 128 / 8;
```
Comment (djmitche) [inline src/api.js]: ??

File: src/api.js [hunk 2]
```diff
@@ -1373,6 +1373,8 @@ api.declare({
     this.workerInfo.seen(provisionerId, workerType, workerGroup, workerId),
   ]);
 
+  await this.workerInfo.taskSeen(provisionerId, workerType, workerGroup, workerId, result);
```
Comment (djmitche) [inline src/api.js]: This should get done in the same Promise.all above, so that it occurs in parallel.

Comment (djmitche) [reply src/api.js]: Although that leaves a bit of a race between workerInfo.seen creating a worker entity and workerInfo.taskSeen updating its list of tasks.  Could this all be included in workerInfo.seen, by passing `tasks` to it?

File: src/api.js (hunk 1 above)
Comment (helfi92) [reply src/api.js]: Oops, unused constant. Will remove.

File: src/api.js (hunk 2 above)
Comment (helfi92) [reply src/api.js]: @djmitche Hm, tasks (`result`) is only available after `this.workClaimer.claim` resolves and so if this happens in parallel to workerInfo.seen, I won't be able to pass `tasks` to it.

Comment (djmitche) [reply src/api.js]: Oh, I didn't see that.  OK, that solves the race, too.

**Pull request merged**